File: MyApplication/app/src/main/java/com/example/myapplication/Activity3.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class Activity3 extends MainActivity {
    @Override
    public void getNextButton() {
        Intent i = new Intent(Activity3.this, Activity4.class );
        startActivity(i);
    }

    @Override
    public void getPreviousButton() {
        Intent i = new Intent(Activity3.this, Activity2.class );
        startActivity(i);
    }

    @Override
    public String getAcitivityName() {
        return "Activity 3";
    }
}

File: MyApplication7/app/src/main/java/com/example/myapplication/Activity3.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_3.*
import kotlinx.android.synthetic.main.activity_main.*

class Activity3 : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_3)

        button4.setOnClickListener {
            startActivity(Intent(this, Activity4::class.java))
        }

        button5.setOnClickListener {
            startActivity(Intent(this, Activity2::class.java))
        }

        supportActionBar!!.title = "Back"
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }
    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
